### ISSUE TITLE
Add FindOMPL.cmake so or_ompl will find ompl package in your catkin workspace

### ISSUE THREAD

**Issue by DavidB-CMU** [personalrobotics/or_ompl#29]
I would like the option to use bleeding edge source of OMPL with or_ompl and prpy.

Therefore I think we should add a `FindOMPL.cmake` to our package `or_ompl`.
Without this file, `or_ompl` will *only* find the official ROS package `ros-indigo-ompl` and uses `OMPL_INCLUDE_DIRS = /opt/ros/indigo/include`

On ROS Answers, people suggest uninstalling `ros-indigo-ompl` and creating a catkin package in your workspace.

However, I found that when I add a `FindOMPL.cmake` file to our package or_ompl, it works for both scenarios without uninstalling anything:

By default, or_ompl will find the ROS package and uses `OMPL_INCLUDE_DIRS = /opt/ros/indigo/include`

Alternatively, if I add a custom `ompl` package in by catkin workspace, `or_ompl` will successfully find it and uses `OMPL_INCLUDE_DIRS = /home/USERNAME/SOME_WORKSPACE/devel/include`

**Comment by mkoval** [comment]
`FindOMPL.cmake` would, ideally, be distributed along with OMPL. Our `CMakeLists.txt` file should be able to find it, even if OMPL was built in a Catkin workspace, as long as OMPL installs the file to the correct directory.

Are you sure it's not included?

**Comment by DavidB-CMU** [comment]
@mkoval You're too quick I was still typing :-) I just revised my first message with more detail.

**Comment by mkoval** [comment]
Ideally, the OMPL repository itself would include a `FindOMPL.cmake` file and install it in the appropriate directory. If this is true, then everything should "just work" and using our own `FindOMPL.cmake` would not be necessary. Can you check if this is the case?

In any case, I'm not sure why including the `FindOMPL.cmake` file ourself fixes the issue. I see two possibilities:

1. We're including a different (and possibly better) `FindOMPL.cmake` file than is included with `ros-indigo-ompl`.
2. The `FindOMPL.cmake` sets the default search path relative to the location of itself

We should verify what is going on before adding the file. It's potentially a brittle change (since the file could change in future versions of OMPL) and, especially in case (2), may break in chained Catkin workspaces.

**Comment by DavidB-CMU** [comment]
@mkoval Yes I can try to debug cmake some more.

If anyone else wants to have a go at installing OMPL in their catkin workspace, here is a catkin package:
https://github.com/DavidB-CMU/OMPL_catkin_pkg

If you want to see the `FindOMPL.cmake` I used, it's here: 
https://github.com/DavidB-CMU/or_ompl/tree/Issue29

**Comment by DavidB-CMU** [comment]
@mkoval You say that "Ideally, the OMPL repository itself would include a FindOMPL.cmake file and install it in the appropriate directory..." but it sounds like going around in circles?

I think the general problem is that catkin or ROS should check my workspace directory *before* it checks /opt/ros/indigo/*".
Is that the expected behaviour?

**Comment by jeking04** [comment]
For what its worth, Jonathan Gammell also struggled with this when he was here a few summers ago. He wanted to have his own (newer) version of ompl and have cmake find it, but he could never get that working. In the end, he just uninstalled the ros version (ros-fuerte-ompl - this was pre-catkin). So it has been a problem for awhile. We could check with him and see if he ever found a better solution.

**Comment by psigen** [comment]
@DavidB-CMU: Thanks for digging into this.  Can you figure out what CMake is using to resolve OMPL when you *don't* provide your `FindOMPL.cmake`?  Maybe that can help answer @mkoval's question about why it's working.

**Comment by DavidB-CMU** [comment]
I can't replicate the problem now, it must've been my mistake. It is sufficient to put an `ompl` package in your workspace and re-build the entire workspace.

In detail:

I created a minimal catkin workspace containing the packages:
ompl
openrave_catkin
or_ompl

I run `catkin build -p 1` which builds them in their dependent order:
ompl
openrave_catkin
or_ompl

And when `or_ompl` calls the `find_package(ompl)` command, it correctly finds the OMPL library in the custom package in my workspace:
`Found OMPL: /home/USERNAME/TEST_WORKSPACE/devel/lib/x86_64-linux-gnu/libompl.so    `

Then if I delete my custom `ompl` package and re-build the entire workspace, `or_ompl` correctly finds the ROS OMPL library:
`Found OMPL: /opt/ros/indigo/lib/x86_64-linux-gnu/libompl.so `

My guess is that, I was building `or_ompl` before my custom `ompl` package, and that's why `find_package(OMPL)` was not finding it! :-)
This time I deleted the *entire* "build" AND "devel" directories in my catkin workspace and re-built the entire workspace, making sure the packages were built in the correct order.
